feat: enhance credential definition parsing for webvh DID format

brianorwhatever opened the pull request:
# Summary of Changes

Addresses #1621. Will now handle getting a name from a `did:webvh` credential definition id

# Screenshots, videos, or gifs

N/A

# Breaking change guide

N/A

# Related Issues

Replace this text with issue #'s that are relevant to this PR. If there are none, simply enter N/A

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this)
- [x] If applicable, screenshots, gifs, or video are included for UI changes
- [x] If applicable, breaking changes are described above along with how to address them
- [ ] If applicable, added [changeset(s)](https://github.com/changesets/changesets)
- [x] Added sufficient [tests](../packages/core/__tests__/) so that overall code coverage is not reduced

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated checks have passed

brianorwhatever committed:
feat: enhance credential definition parsing for webvh DID format

Signed-off-by: Brian Richter <brian@aviary.tech>

brianorwhatever committed:
refactor: replace parseCredDefFromId with getCredentialName for improved credential name retrieval

- Updated tests to reflect changes in credential name retrieval logic.
- Enhanced CredentialDetails and CredentialOffer components to use async getCredentialName function.
- Refactored utility functions to support new credential name retrieval method, ensuring compatibility with both webvh and Indy formats.

Signed-off-by: Brian Richter <brian@aviary.tech>

bryce-mcmath commented on packages/core/src/utils/cred-def.ts:
```diff
@@ -1,9 +1,43 @@
+      const result: AnonCredsCredentialDefinition = await agent.modules.anoncreds.getCredentialDefinition(agent.context, credDefId)
+      name = result.tag
+    } catch {
+      // If the credential definition is not found, return the default name
```
```suggestion
      agent.config.logger('parseWebVHCredDefId: Credential definition not found, using default name')
```

bryce-mcmath commented on packages/core/src/utils/cred-def.ts:
```diff
@@ -1,9 +1,43 @@
+      const result: AnonCredsSchema = await agent.modules.anoncreds.getSchema(agent.context, schemaId)
+      name = result.name
+    } catch {
+      // If the schema is not found, return the default name
```
```suggestion
      agent.config.logger('parseWebVHCredDefId: Schema definition not found, using default name')
```

bryce-mcmath commented on packages/core/src/utils/cred-def.ts:
```diff
@@ -1,9 +1,43 @@
+    }
+  }
+
+  if ((name.toLocaleLowerCase() === 'default' || name.toLowerCase() === 'credential') && schemaId) {
```
Just curious, why different lower case methods here?

brianorwhatever replied:
this was copied from elsewhere. I have updated all of them to use regular `.toLowerCase`

bryce-mcmath reviewed (changes requested):
Some minor nits. Have you tested this with the Bifold sample app or BC Wallet? Using the showcase for example

brianorwhatever committed:
chore(tests): update Jest mocks for React Native modules

- Refactored Jest mocks for 'react-native-splash-screen', 'react-native-toast-message', and 'react-native-config' to use virtual module support.
- Cleaned up the test setup file to fix errors.

Signed-off-by: Brian Richter <brian@aviary.tech>

brianorwhatever committed:
Apply suggestion from @bryce-mcmath

Co-authored-by: Bryce McMath <32586431+bryce-mcmath@users.noreply.github.com>
Signed-off-by: Brian Richter <brian@aviary.tech>

brianorwhatever committed:
Apply suggestion from @bryce-mcmath

Co-authored-by: Bryce McMath <32586431+bryce-mcmath@users.noreply.github.com>
Signed-off-by: Brian Richter <brian@aviary.tech>

brianorwhatever committed:
refactor: standardize logger usage and improve string comparison in credential definition utilities

- Updated logger calls to use the `info` method for better logging consistency.
- Refactored string comparison to use `toLowerCase()` instead of `toLocaleLowerCase()` for uniformity across credential definition parsing functions.

Signed-off-by: Brian Richter <brian@aviary.tech>

brianorwhatever committed:
test: enhance credential name retrieval tests for webvh and Indy formats

- Added new test cases to validate the behavior of `getCredentialName` under various scenarios, including agent configurations and error handling.
- Introduced tests for `parsedCredDefName` and `parsedCredDefNameFromCredential` to ensure correct name resolution from credential metadata.

Signed-off-by: Brian Richter <brian@aviary.tech>

bryce-mcmath committed:
docs(changeset): enhance credential definition parsing for webvh

Signed-off-by: Bryce McMath <32586431+bryce-mcmath@users.noreply.github.com>

brianorwhatever committed:
Merge pull request #1 from bryce-mcmath/patch-1

docs(changeset): enhance credential definition parsing for webvh
